_posts/2019-11-04-css-is-hard-and-vanguard-2019-learnings.md

```diff
@@ -0,0 +1,211 @@
+communicating early and often when there’s any uncertainty around delivering a feature by deadline is critical, and
+working with team members that value candor, flexibility, communication, and iteration is empowering.
+
+## **3. CSS is Hard**
+
+I get excited about implementing visually delightful designs, and Artsy Vanguard’s 2019 designs did not disappoint.
+There were two key components that involved interesting CSS approaches:
+
+### - Mixed Blending Mode to handle the inversion of the side frame text when it appeared over different backgrounds.
+
+### - Creating the visual effect of `.mp4` videos nested inside of SVG shapes
+
+**_Mixed Blend Mode_**
+
+Mixed blend mode is a CSS property that handles blending the colors of overlapping elements. In this case, we
+wanted to create the effect of inverting the colors of the static side frame text when they overlapped with either
+the white background, the black SVG shapes, or the SVG videos.
+
+This inverted color effect also needed to be applied to the text of the editorial content when it overlapped with
+the black SVG background, when it extended past the SVG background onto a white background, and when the current
+article was in an expanded state, such as when a user selected “Read More”.
+
+I found [CSS-Tricks](https://css-tricks.com/almanac/properties/m/mix-blend-mode/) and the
+[Mozilla Developer’s Network](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode) documentation super
+helpful in understanding the various blending properties. Since the designs for Vanguard involved a fairly simple
+blending mode of inverting black and white, the mix-blend-mode difference property was how we created the effect of
+inverting the black text against the various backgrounds it overlapped with. The blend mode property works by
+calculating the rgb color value of the text, and the background-color of the underlying element, and using that
+value to determine what colors to render. This inverse calculation also creates the “solarized” effect when the
+black side frame text renders over the video-in-SVG.
+
+In the original design specifications, the static text framing the page was intended to render black over a white
+background, white over a black background, and white over the video-in-SVG. In order to achieve that effect, we
+would have needed to apply the `difference` mix-blend-mode property when the text overlapped with a black or white
+background, and color-dodge when it overlapped with a video-in-SVG background. However, given it is not possible to
+assign multiple blending modes to a fixed element we had to deviate from the original design specifications.
+
+![](https://lh6.googleusercontent.com/woJxTfbbVj4-Ev_Aor2oskc05skM0GFu88G7AofeCILJpW_Q_okeXT993CLdod4rvr_mSA2BcUnTL7MDpssJlAxl79roF0ssMR_pPY0az7LPof-8VfQ5kqW1VakcYtIt7HqeIFFq)
+
+```
+const  FrameText  =  styled(Sans)
+position: fixed;
```

Comment:
I think the styled components are all missing backticks surrounding their styles. So instead of...

```
... = styled(Sans)
styles: here;
```

They should be...

```
... = styled(Sans)`
styles: here;
`
```

Reply:
I think adding the formating suffix causes the editor to no longer respect the backticks. Looks like it still renders as a codeblock, however.